packages/studio/src/store/slices/pages/createPageSlice.ts

```diff
@@ -158,6 +163,29 @@ export const createPageSlice: SliceCreator<PageSlice> = (set, get) => {
         (component) => component.uuid === activeComponentUUID
       );
     },
+    getFirstSelectedComponentUUID: () => {
+      const { selectedComponentUUIDs, getActivePageState } = get();
+      const activePageState = getActivePageState();
+      const componentTree = activePageState?.componentTree;
+      if (!componentTree) {
+        throw new Error(
+          `Error rendering components: component tree is undefined.`
+        );
+      }
+      const orderedSelectedComponentUUIDs =
+        ComponentTreeHelpers.mapComponentTreeParentsFirst(
+          componentTree,
+          (c) => {
+            if (selectedComponentUUIDs.has(c.uuid)) {
+              return c.uuid;
+            }
+            return undefined;
+          }
+        ).filter((uuid) => !!uuid);
```

Comment:
nit: I think the mapping logic could be a little simpler if we filtered to the selected components first

Reply:
nvm, I think that could be problematic since `mapComponentTreeParentsFirst` wouldn't necessarily have the correct `parentUUID` to look for within the selected components unless you compute that first and pass it in